sstables/sstable_set.cc

```diff
@@ -172,14 +172,14 @@ stop_iteration sstable_set::for_each_sstable_until(std::function<stop_iteration(
     return _impl->for_each_sstable_until(std::move(func));
 }
 
-void
+bool
 sstable_set::insert(shared_sstable sst) {
-    _impl->insert(sst);
+    return _impl->insert(sst);
 }
 
-void
+bool
 sstable_set::erase(shared_sstable sst) {
-    _impl->erase(sst);
+    return _impl->erase(sst);
```

Comment:
I understand how this came about, but can't we prevent it? This will infect the API.

I can't think of a way to prevent it. If we had on_internal_error() since the beginning it would be possible, but adding them now is difficult.

Reply:
Right.  I don't think it's a common case, but I can't say for sure erase would only be called for sstables that exist in the set.